engine/src/main/java/org/terasology/utilities/subscribables/SpecificSubscribable.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.utilities.subscribables;
+
+import java.beans.PropertyChangeListener;
+
+/**
+ * Originally created to provide Observable-like functionality to config classes,
+ * it can be used by any class that wishes to be monitored for change in specific properties.
+ */
+public interface SpecificSubscribable {
```

Comment:
Short answer: if you really want me to. 0:)
Longer answer: what I was aiming for is to create a flexible solution, as some objects might need to be monitored on a per-property basis, some objects might need to be monitored as a whole and some objects might need to be monitored both ways. As there is no multiple inheritance in java I decided to provide all three possible combinations of abstract base class plus the two interfaces that can be composed together. Why is this not a good approach?